apps/drupal-default/particle_theme/templates/paragraphs/image-hover/paragraph--hover-images.html.twig

```diff
@@ -0,0 +1,75 @@
+ *   - paragraph.getOwnerId(): The user ID of the paragraph author.
+ *   See Drupal\paragraphs\Entity\Paragraph for a full list of public properties
+ *   and methods for the paragraph object.
+ * - content: All paragraph items. Use {{ content }} to print them all,
+ *   or print a subset such as {{ content.field_example }}. Use
+ *   {{ content|without('field_example') }} to temporarily suppress the printing
+ *   of a given child element.
+ * - attributes: HTML attributes for the containing element.
+ *   The attributes.class element may contain one or more of the following
+ *   classes:
+ *   - paragraphs: The current template type (also known as a "theming hook").
+ *   - paragraphs--type-[type]: The current paragraphs type. For example, if the paragraph is an
+ *     "Image" it would result in "paragraphs--type--image". Note that the machine
+ *     name will often be in a short form of the human readable label.
+ *   - paragraphs--view-mode--[view_mode]: The View Mode of the paragraph; for example, a
+ *     preview would result in: "paragraphs--view-mode--preview", and
+ *     default: "paragraphs--view-mode--default".
+ * - view_mode: View mode; for example, "preview" or "full".
+ * - logged_in: Flag for authenticated user status. Will be true when the
+ *   current user is a logged-in member.
+ * - is_admin: Flag for admin user status. Will be true when the current user
+ *   is an administrator.
+ *
+ * @see template_preprocess_paragraph()
+ *
+ * @ingroup themeable
+ */
+#}
+{#
+    A minimum of three twig files are necessary in using paragraphs with blocks. The block-specific twig file, the block-field-specific twig file, and the paragraph-specific twig file.
+    In this file we add a class to execute functionality, add the appropriate markup, and get the raw value from the text and link fields.
+#}
+
+{% block paragraph %}
+
+  {% block content %}
+
+    {% set image_link = content.field_hover_link.0['#url'] %}
+
+          {{ content.field_hover_image.0 }}
+          {% if image_link is null %}
+            <figcaption class="absolute block left-0 right-0 w-full cursor-default">
+              <div class="utc-border-wrapper">
+                <h2 class="m-0 lg:mb-2 font-bold capitalize text-utc-new-blue-500 leading-tight">{{ content.field_hover_title.0 }}</h2>
+                <p class="m-0 mb-2 text-utc-new-blue-500 leading-tight">{{ content.field_hover_text.0 }}</p>
+              </div>
+            </figcaption>
+          {% else %}
+            <figcaption class="absolute block left-0 right-0 w-full cursor-pointer">
+              <div class="utc-border-wrapper">
+                <h2 class="m-0 lg:mb-2 font-bold capitalize text-utc-new-blue-500 leading-tight">{{ content.field_hover_title.0 }}</h2>
+                <p class="m-0 mb-2 text-utc-new-blue-500 leading-tight">{{ content.field_hover_text.0 }}</p>
+                <p class="hover-image-space">&nbsp;</p>
```

Comment:
why do we have an empty p tag?

Reply:
Because the anchor on the mobile turns into a button link, instead of the whole text area being the link, and I need to add space between the text and the button. The margin-bottom in the paragraph above won't apply here (and is removed in my new files), because of the transitions. I can remove it and use another solution.